app/controllers/settings/two_factor_authentication_methods_controller.rb

```diff
@@ -4,14 +4,12 @@ module Settings
   class TwoFactorAuthenticationMethodsController < BaseController
     include ChallengableConcern
 
-    layout 'admin'
+    skip_before_action :require_functional!
 
-    before_action :authenticate_user!
+    before_action :require_not_suspended!
```

Comment:
As I said in my previous comment, I think it's probably fine to allow changing 2FA settings for suspended users.